### ISSUE TITLE
Incorrect security.protocol configuration or broker version

### ISSUE THREAD

**Issue by 0ni0nrings** [confluentinc/confluent-kafka-python#1035]
Description
===========

Using kafka python library to create a topic programmatically. Removing AdminClient & creating topic beforehand results in successful producer run. Adding AdminClient brings error to the setup as shown below.

```
>>> a = AdminClient({'bootstrap.servers': 'pkc-ldvj1.ap-southeast-2.aws.confluent.cloud:9092'})

>>> %6|1613615645.328|FAIL|rdkafka#producer-1| [thrd:pkc-ldvj1.ap-southeast-2.aws.confluent.cloud:9092/bootstrap]: pkc-ldvj1.ap-southeast-2.aws.confluent.cloud:9092/bootstrap: Disconnected while requesting ApiVersion: might be caused by incorrect security.protocol configuration (connecting to a SSL listener?) or broker version is < 0.10 (see api.version.request) (after 50ms in state APIVERSION_QUERY)
%6|1613615645.428|FAIL|rdkafka#producer-1| [thrd:pkc-ldvj1.ap-southeast-2.aws.confluent.cloud:9092/bootstrap]: pkc-ldvj1.ap-southeast-2.aws.confluent.cloud:9092/bootstrap: Disconnected while requesting ApiVersion: might be caused by incorrect security.protocol configuration (connecting to a SSL listener?) or broker version is < 0.10 (see api.version.request) (after 68ms in state APIVERSION_QUERY, 1 identical error(s) suppressed)
```

How to reproduce
================

Start Python interpreter
```
>>> from confluent_kafka.admin import AdminClient, NewTopic
>>> a = AdminClient({'bootstrap.servers': 'pkc-ldvj1.ap-southeast-2.aws.confluent.cloud:9092'})
```

Checklist
=========
Please provide the following information:

 - [ ] confluent-kafka-python and librdkafka version (`confluent_kafka.version()` and `confluent_kafka.libversion()`):
 ```
 Name: confluent-kafka
Version: 1.6.0
Summary: Confluent's Python client for Apache Kafka
Home-page: https://github.com/confluentinc/confluent-kafka-python
Author: Confluent Inc
Author-email: support@confluent.io
License: UNKNOWN
Location: /Users/ccloud_example/lib/python3.8/site-packages
Requires: 
Required-by: 
```
 - [ ] Apache Kafka broker version:
 - [ ] Client configuration: `{...}`
 - [ ] Operating system: (MacOS & CentOS)
 - [ ] Provide client logs (with `'debug': '..'` as necessary)
 - [ ] Provide broker log excerpts (managed platform)
 - [ ] Critical issue

**Comment by edenhill** [comment]
In the CCloud UI, look at the client configuration tab for librdkafka-based clients.
You'll see that you also need to specify security.protocol=SASL_SSL, sasl.mechanism=PLAIN, sasl.username=yourapikey, sasl.password=yoursecret, etc.

**Comment by Dipenbhatt03** [comment]
Is this issue still there, facing the same when using confluent kafka brokers and connecting from python faust.

**Comment by edenhill** [comment]
This is not an issue with the client, but missing configuration passed by the user.

**Comment by subasah** [comment]
I am trying to connect via C Sharp code with a docker image running `debezium/kafka:1.7` and getting this error message: 

`localhost:29092/bootstrap: Disconnected while requesting ApiVersion: might be caused by incorrect security.protocol configuration (connecting to a SSL listener?) or broker version is < 0.10 (see api.version.request) (after 1ms in state APIVERSION_QUERY)`

Any help will be appreciated!
Thanks

**Comment by tsawyer999** [comment]
Hello @subasah !

Curiosity, is it possible to get a follow up how you resolve it?

I am facing the same issue.

Thank you

**Comment by RobScott-Trimble** [comment]
Ditto, exact same thing here...  Runs fine and I'm able to consume from a Kafka Topic _locally_, but as soon as I deploy to Kubernetes Docker container, I get that error over and over..

**Comment by molayodecker** [comment]
> Disconnected while requesting ApiVersion: might be caused by incorrect security.protocol configuration

If this is happening in local, destroy and pull a fresh copy of the images

**Comment by iamaslowwalker** [comment]
> Ditto, exact same thing here... Runs fine and I'm able to consume from a Kafka Topic _locally_, but as soon as I deploy to Kubernetes Docker container, I get that error over and over..

I was facing the same issue, use the below config to resolve it:
config = {
                'bootstrap.servers': "your_broker_config', 
                'default.topic.config': {'api.version.request': True}, 
                'security.protocol': 'SSL',
            }
    producer = Producer(config)

**Comment by Durga1prasad** [comment]
does anyone solution for this?

**Comment by uncommonthinker** [comment]
> Hi guys this is my solution, works fine for me and have faced the same issue: edit kafka's configuration file
> 
> 1. set listener to `PLAINTEXT://0.0.0.0:9092`
> 2. set advertised.listeners to PLAINTEXT://your.host.name:9092
>    notice : your.host.name is the address which your client could access to
>    for example : my kafka server private ip 172.16.0.1, public ip 6.6.6.6,  then 6.6.6.6 is the value of your.host.name


Getting the below exception,
java.lang.IllegalArgumentException: requirement failed: advertised.listeners cannot use the nonroutable meta-address 0.0.0.0. 
Use a routable IP address.

**Comment by NSmolianitski** [comment]
Just solved this error by changing localhost to 0.0.0.0 in KAFKA_LISTENERS in my docker-compose file as it is shown in [confluent docs](https://docs.confluent.io/platform/current/installation/docker/config-reference.html#cp-kafka-example). Note that KAFKA_ADVERTISED_LISTENERS should still use localhost.

My example docker-compose.yaml:
```yaml
services:
    kafka:
        image: confluentinc/cp-kafka:latest
        volumes:
            - ./kafka-persistence:/var/lib/kafka/data
        ports:
            - "9092:9092"
        environment:
            CLUSTER_ID: "hello_world"
            KAFKA_NODE_ID: 1
            KAFKA_PROCESS_ROLES: broker,controller
            KAFKA_OFFSETS_TOPIC_REPLICATION_FACTOR: 1
            KAFKA_CONTROLLER_LISTENER_NAMES: "CONTROLLER"
            KAFKA_CONTROLLER_QUORUM_VOTERS: "1@kafka:29093"
            KAFKA_LISTENER_SECURITY_PROTOCOL_MAP: "CONTROLLER:PLAINTEXT,PLAINTEXT:PLAINTEXT,PLAINTEXT_HOST:PLAINTEXT"
            KAFKA_LISTENERS: "PLAINTEXT://kafka:29092,CONTROLLER://kafka:29093,PLAINTEXT_HOST://0.0.0.0:9092"
            KAFKA_ADVERTISED_LISTENERS: "PLAINTEXT://kafka:29092,PLAINTEXT_HOST://localhost:9092"
```

**Comment by rneha725** [comment]
Above resolution worked for me.
